### PR TITLE
fix(1010): [small] update the workflowGraph of the event with prChain enabled.

### PR DESCRIPTION
**Pull request by wdstar**
## Context

Workflow graphs of PR chain enabled event have no job ids.
This PR adds job ids to those graphs.
And I add `rewire` package for unit tests. Let's write unit tests for each function!

## Note

- Currently `PR-#:` prefixed name is not allowed by the workflow graph validator. We should consider the addition of the prefix carefully.
```
[request,server,error] data: Error: child "workflowGraph" fails because [child "nodes" fails because ["nodes" at position 2 fails because [child "name" fails because ["name" with value "PR-1:first-job" fails to match the required pattern: /^~([\w-]+)$/, "name" with value "PR-1:first-job" fails to match the required pattern: /^[\w-]+$/, "name" with value "PR-1:first-job" fails to match the required pattern: /^~(sd@\d+:[\w-]+|(pr|commit)(:(.+))?)$/]]]]
    at postValidate (/usr/src/app/node_modules/hapi/lib/validation.js:214:30)
...
```

## References

- https://github.com/screwdriver-cd/screwdriver/issues/1010#issuecomment-462615428

### CODE REVIEW

File: lib/eventFactory.js [hunk 1]
```diff
@@ -265,6 +265,48 @@ function getLatestWorkflowGraph(config) {
     return Promise.resolve(pipeline.workflowGraph);
 }
 
+/**
+ * Update the workflowGraph
+ * @method updateWorkflowGraph
+ * @param  {Object}         config
+ * @param  {Pipeline}       pipeline            Pipeline
```
Comment (tk3fftk) [inline lib/eventFactory.js]: These JSDoc of params should be `config.*`

Comment (wdstar) [reply lib/eventFactory.js]: I will fix it. Thanks.

Comment (wdstar) [reply lib/eventFactory.js]: @tk3fftk fixing done.

Comment (tk3fftk) [reply lib/eventFactory.js]: Sorry, my intention is `config.pipeline`, `config.eventConfig`... like other JSDoc.

File: lib/eventFactory.js [hunk 2]
```diff
@@ -265,6 +265,48 @@ function getLatestWorkflowGraph(config) {
...
+ * @param  {Number}         pipeline.id         Pipeline id
+ * @param  {Object}         eventConfig.*
+ * @param  {Number}         [eventConfig.prNum] PR number
+ * @param  {Object}         workflowGraph       WorkflowGraph
+ * @resolves {Object}                           Resolves with workflowGraph
+ */
+function updateWorkflowGraph(config) {
+    const { pipeline, eventConfig, workflowGraph } = config;
+
+    if (pipeline.prChain) {
```
Comment (yuichi10) [inline lib/eventFactory.js]: Don't you need to check `eventConfig.prRef` here?
Only event PR has prChain?

File: lib/eventFactory.js (hunk 1 above)
Comment (wdstar) [reply lib/eventFactory.js]: OK, I copied the existing `getLatestWorkflowGraph()`'s JSDoc, which was invalid.

File: lib/eventFactory.js (hunk 2 above)
Comment (wdstar) [reply lib/eventFactory.js]: No, we must check the base branch (`pipeline` object)'s `prChain` flag.
Please see https://github.com/screwdriver-cd/models/pull/332

**Review by tk3fftk — APPROVED** [review]
LGTM:+1:

File: lib/eventFactory.js (hunk 2 above)
Comment (yuichi10) [reply lib/eventFactory.js]: When the event come from commit, is there possibility the `pipeline.prChain` has true? 
If so maybe you also need to check the event is came from pr.
Then the if sentense will be `if (eventConfig.prRef && pipline.prChain)`.

Comment (sakka2) [reply lib/eventFactory.js]: I agree with `if (eventConfig.prRef && pipline.prChain)` .
We need check if the event is pr or not.

Comment (wdstar) [reply lib/eventFactory.js]: @yuichi10 Your advice is very correct. I fixed it. Thanks!

**Review by yuichi10 — APPROVED** [review]
LGTM

**Review by sakka2 — APPROVED** [review]
LGTM!

**Review by tkyi — APPROVED** [review]
Wondering if it would be better to combine `getLatestWorkflowGraph` and `updateWorkflowGraph`, since they take in similar arguments

Otherwise LGTM 👍

**Comment by wdstar** [comment]
> Wondering if it would be better to combine `getLatestWorkflowGraph` and `updateWorkflowGraph`, since they take in similar arguments

@tkyi We should keep a function simple and let it to do one thing well. If it is not so, it will be more difficult to test a function.

**Review by minzcmu — APPROVED** [review]

**Pull request merged**